app/jobs/check_job.rb

```diff
@@ -0,0 +1,25 @@
+class CheckJob < ApplicationJob
+  queue_as :default
+
+  def perform(check, batch: nil, callback_uri: nil)
+    return if check.started_at || check.completed_at
+
+    check.update!(started_at: Time.now)
+
+    report = LinkChecker.new(check.link.uri).call
+
+    check.update!(
+      link_errors: report.errors,
+      link_warnings: report.warnings,
+      completed_at: Time.now
+    )
+
+    if callback_uri
+      if batch
+        WebhookJob.perform_now(batch, callback_uri) if batch.completed?
```

Comment:
I think we probably want to hit the callback each time there is progress, that'd allow apps to gradually update their ui rather than waiting for the whole thing?